src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/ADAMInstance.cs

```diff
@@ -144,7 +144,7 @@ public static AdamInstance GetAdamInstance(DirectoryContext context)
                 {
                     throw new ActiveDirectoryObjectNotFoundException(SR.Format(SR.AINotFound, context.Name), typeof(AdamInstance), context.Name);
                 }
-                dnsHostName = (string)PropertyManager.GetPropertyValue(context, rootDSE, PropertyManager.DnsHostName);
+                dnsHostName = (string)PropertyManager.GetPropertyValue(context, rootDSE, PropertyManager.DnsHostName)!;
```

Comment:
Note it will only throw if property value is not there at all but if null value is present it will not throw and return null instead. Also notes that `Properties` is public and while there is a code path guarding against null in `DirectoryEntry.Properties[anyName].Value` (it's a no-op) then there is nothing guarding against `DirectoryEntry.Properties[anyName].Add(null)`.

Reply:
You are right, the code is not guaranteeing against null, but somehow [PropertyValueCollection[index]](https://docs.microsoft.com/en-us/dotnet/api/system.directoryservices.propertyvaluecollection.item?view=dotnet-plat-ext-5.0) and [PropertyValueCollection.Add(object value)](https://docs.microsoft.com/en-us/dotnet/api/system.directoryservices.propertyvaluecollection.add?view=dotnet-plat-ext-5.0#exceptions) both documented to be non null. Literally for all references we are suppressing the return value from `(string)PropertyManager.GetPropertyValue(context, rootDSE, PropertyManager.DnsHostName)!` which also illustrates the intention was `non-null`, there is something wrong here or we are missing something.

Reply:
With that being said I am resolving this and all comments related to this, please take a look at other unrelated comments

Reply:
>  there is nothing guarding against DirectoryEntry.Properties[anyName].Add(null)

Even it is allowing null I think setting it into null should be an error, not sure who is the expert in this area, I am OK leaving this nullable for this PR but i think we should file an issue for this

Reply:
https://github.com/dotnet/runtime/issues/49128